### ISSUE TITLE
Node List pollution

### ISSUE THREAD

**Issue by simon1tan** [K-3D/k3d#33]
The node list is overcrowded. This list should be separated with user created objects and built-in objects. Scrolling through 20 extra objects just to find my what I'm looking for is no good.

**Comment by barche** [comment]
Yes, there are a lot of nodes. I think implementing various filters for this list is a great idea, but unfortunately I don't really have the time to implement new features at the moment. You are more than welcome to take a stab at it yourself, the relevant source file is here:

https://github.com/K-3D/k3d/blob/master/modules/ngui_node_list_panel/panel.cpp

**Comment by Megalomaniak** [comment]
On a related note what I'm finding really confusing is that the Node List(which far as I can tell is the closest thing to a scene outliner in other 3D content creation suites) is rendered as a list instead of a tree hierarchy and the node history which is essentially a sort of modifier(procedures/operations) stack is rendered as a tree hierarchy.

**Comment by barche** [comment]
The reason for the flat list is that the list currently is a mix of different kind of nodes, for example transform modifiers work on the transformation matrices, while mesh modifiers work on the mesh data. This makes it difficult to uniquely define which node is a child and which is a parent. The proper representation is actually a directed graph, but implementing nice GUI for that is a lot of work.

**Comment by Megalomaniak** [comment]
With Qt I know that nodeeditor can be used to create graph interfaces.
https://github.com/paceholder/nodeeditor

Not sure if something similar exists for GTK+. nvm. Ask google and google provides:
https://github.com/GNOME/graph-gtk
